AtelierTomato.Markov/AtelierTomato.Markov.Generation/MarkovChain.cs

```diff
@@ -1,24 +1,35 @@
 ﻿using AtelierTomato.Markov.Data;
 using AtelierTomato.Markov.Data.Model;
+using AtelierTomato.Markov.Data.Model.ObjectOID;
 using Microsoft.Extensions.Options;
 
 namespace AtelierTomato.Markov.Generation
 {
-	public class GenerateMarkovSentence(ISentenceAccess sentenceAccess, IOptions<MarkovGenerationOptions> options)
+	public class MarkovChain(ISentenceAccess sentenceAccess, IOptions<MarkovGenerationOptions> options)
 	{
 		private readonly ISentenceAccess sentenceAccess = sentenceAccess;
 		private readonly MarkovGenerationOptions options = options.Value;
 		private static readonly Random random = new();
 
-		public async Task<string> Generate(SentenceFilter filter)
+		public async Task<string> Generate(SentenceFilter filter, string? firstWord = null)
 		{
-			Sentence? sentence = await GetFirstSentence(filter) ?? throw new Exception("Couldn't query any messages.");
-			string firstWord = sentence.Text.Substring(0, sentence.Text.IndexOf(' '));
-			var tokenizedSentence = new List<string> { firstWord };
-			var prevList = tokenizedSentence;
-
 			// Tracks the IDs of previously used sentences so that we don't recreate an existing sentence or ping pong between two sentences.
-			var prevIDs = new List<string> { sentence.OID.ToString() };
+			List<IObjectOID> prevIDs = [];
+			Sentence? sentence;
+			if (firstWord is null)
+			{
+				sentence = await GetFirstSentence(filter) ?? throw new Exception("Couldn't query any messages.");
+				firstWord = sentence.Text.Substring(0, sentence.Text.IndexOf(' '));
+				prevIDs.Add(sentence.OID);
+			} else
+			{
+				// We need the prevIDs list to not be empty, but if a firstWord is given, it will be empty on first run, and could potentially fail to query.

```

Comment:
wait, why? can't we make the places the consume it work correctly with empty lists? working around it here seems bad.

Reply:
Yeah, probably I just didn't bother at the time, it might be possible.